modules/aws/sonar-upgrader/python_upgrader/tests/test_main.py

```diff
@@ -282,15 +287,15 @@ def test_main_upgrade_failure_with_stop_on_failure_true(args, test_connection_mo
     # then
     assert test_connection_mock.call_count == 3
     call_args_list = run_remote_script_mock.call_args_list
-    assert len(call_args_list) == 7
+    assert len(call_args_list) == 4
+    assert collect_facts_mock.call_count == 3
     for host in ["host1", "host2", "host100"]:
-        assert count_remote_calls_with_host_and_script(call_args_list, host, "get_python_location.sh") == 1
         assert count_remote_calls_with_host_and_script(call_args_list, host, "run_preflight_validations.py") == 1
     assert count_remote_calls_with_host_and_script(call_args_list, "host1", "upgrade_v4_10.sh") == 1
 
 
 def test_main_python_location_failure_with_stop_on_failure_false(
```

Comment:
This test_main_python_location_failure_with_stop_on_failure_false according to its name supposed to check that in case the first gw fails to get the python location, i.e. failed to collect node info), and the stop_on_failure=False, then the code continue to the rest of the nodes (currently there is an issue about this case in the main.py, please see the other comment about the return statement. So in this test you should simulate it according to the new code, i.e. mock collect node info to return some error.